shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Metadata.hs

```diff
@@ -63,6 +74,17 @@ data Metadata = Metadata'
   deriving (Eq, Show, Generic)
   deriving (NoThunks) via AllowThunksIn '["mdBytes"] Metadata
 
+instance SafeToHash Metadata where
+  originalBytes = LBS.toStrict . mdBytes
```

Comment:
This is somewhat confusing when the comments above say "we never add other instances of this except by deriving" (or words to that effect!)

I understand why we do this here, but the comment in the `Hashing` file should be changed, and you should perhaps have a comment here saying "We do this because the legacy Shelley types store their bytes in a different way, and thus we cannot rely on our default instances, yada yada yada" or something like that

Reply:
Your 100% right about it being confusing, the comments now read.`====================================================================
{- | Only Types that preserve their serialisation bytes are members of the
 class SafeToHash. There are only a limited number of primitive direct
 instances of SafeToHash, all but two of them are present in this file. Instead
 of making explicit instances, we almost always use a newtype (around a type S)
 where their is already an instance (SafeToHash S). In that case the newtype
 has its SafeToHash instance derived using newtype deriving. The only exceptions
 are the legacy Shelley types Metadata and Tx that preserve their serialisation bytes
 using a different mechanism than MemoBytes.  SafeToHash is a superclass
 requirement of the classes HashAnnotated and HashWithCrypto (below) which
 provide more convenient ways to construct SafeHashes than using makeHashWithExplicitProxys.`

Reply:
I also added comments for the two explicit instances on Metadata and Tx.